### PR TITLE
Add W91 pwm drivers

### PR DESCRIPTION
**Pull request by UR6LAL**
Added PWM drivers implementation for W91 platform

### CODE REVIEW

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi [hunk 1]
```diff
@@ -113,3 +125,9 @@
 	pinctrl-0 = <&i2c_scl_default &i2c_sda_default>;
 	pinctrl-names = "default";
 };
+
+&pwm0 {
+	status = "okay";
+	clock-frequency = <1000000>;
```
Comment (andriy-bilynskyy) [inline boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi]: Where do we set this frequency?
Is it some constant value? If yes set it in the yaml.

File: drivers/pwm/pwm_w91.c [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+#include <zephyr/drivers/pwm.h>
+#include <ipc/ipc_based_driver.h>
+
+#define LOG_LEVEL CONFIG_PWM_LOG_LEVEL
+LOG_MODULE_REGISTER(pwm_telink);
+
+/* Driver dts compatibility: telink,w91_pwm */
+#define DT_DRV_COMPAT telink_w91_pwm
+
+#define MAX_CYCLE 1638u
```
Comment (andriy-bilynskyy) [inline drivers/pwm/pwm_w91.c]: Is it value from HW? usually this value is 2^n

File: drivers/pwm/Kconfig.w91 [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	default y
+	depends on DT_HAS_TELINK_W91_PWM_ENABLED
+	help
+	  Enables Telink W91 PWM driver.
+
+if PWM_TELINK_W91
+
+config PWM_TELINK_W91_IPC_RESPONSE_TIMEOUT_MS
+	int "Telink IPC dispatcher timeout while waiting from the endpoint response"
+	default 100
```
Comment (andriy-bilynskyy) [inline drivers/pwm/Kconfig.w91]: Do we really need such value? Most drivers have 10

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi (hunk 1 above)
Comment (UR6LAL) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi]: This value from FREERTOS core, but I've just found that we can use API to get this value. I will rework this part

File: drivers/pwm/Kconfig.w91 (hunk 3 above)
Comment (UR6LAL) [reply drivers/pwm/Kconfig.w91]: I will try to use 10 ms

File: drivers/pwm/pwm_w91.c (hunk 2 above)
Comment (UR6LAL) [reply drivers/pwm/pwm_w91.c]: This value isdescribed in "tlsr_timer.h", but there is no define for this:
struct tlsr_timer_cfg_pwm {
	uint16_t high; /* usec, up to ~1638 usec (UINT16_MAX / 40MHz) */
	uint16_t low;  /* usec, up to ~1638 usec (UINT16_MAX / 40MHz) */
	uint8_t park;  /* level of PWM signal when the timer is stopped */
};

File: boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi (hunk 1 above)
Comment (andriy-bilynskyy) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi]: as I see the situation is the same...
set It to const, othewise changing this value will fail the driver.

Comment (UR6LAL) [reply boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi]: I don't use this value anymore, I just forgot to remove it

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**